src/titiler/core/titiler/core/factory.py

```diff
@@ -812,6 +824,299 @@ def wmts(
+
+            """
+            request_args = parse_qsl(str(request.url).split('?')[1])
+
+            out_wms_params = {'layers':[]}
+            for param in request_args:
+                key = param[0].lower()
+                
+                if key not in base_wms_params:
+                    continue
+
+                value = param[1]
+
+                if key in ("height", "width"):
+                    value = int(value)
+                    
+                elif key in ["crs", "srs"]:
+                    key = "srs"
+                    out_wms_params["dst_crs"] = value
+                    out_wms_params["bounds_crs"] = value
+                    
+                elif key == "bbox":
+                    value = [float(xy) for xy in value.split(",")]
+                    if len(value) != 4:
+                        raise HTTPException(
+                            status_code=400,
+                            detail="bbox needs 4 coordinates separated by commas",
+                        )
+                elif key == "version" and value not in set(WMSVersion):
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Version specified not allowed. Allowed versions include: {[w.value for w in list(WMSVersion)]}",
+                    )
+                elif key == 'format':
+                    value = WMSMediaType(value)
+                    
+                if key == "layers":
+                    out_wms_params[key] += [value]
+                else:
+                    out_wms_params[key] = value
+                    
+            if out_wms_params.get("bbox"):
+
+                # WMS 1.3.0 is lame and flips the coords of EPSG:4326
+                if (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs")
+                    and out_wms_params.get("srs") == "EPSG:4326"
+                ):
+                    out_wms_params["bbox"] = [
+                        out_wms_params["bbox"][1],
+                        out_wms_params["bbox"][0],
+                        out_wms_params["bbox"][3],
+                        out_wms_params["bbox"][2],
+                    ]
+                    
+                # Overwrite CRS:84 with EPSG:4326 when specified
+                elif (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs") == "CRS:84"
+                ):
+                    out_wms_params["srs"] = "EPSG:4326"
+                    
+                # `bounds` is compatible with reading from rio.io.reader
+                out_wms_params['bounds'] = out_wms_params['bbox']
+                    
+            print(out_wms_params)
+            return out_wms_params
+
+        @self.router.get("/wms")
+        def wms(
+            _request: Request,
+            request: WMSRequestType = "GetCapabilities",
+            version: WMSVersion = "1.1.1",
```

Comment:
I've been using `Literal` over enums lately. It gives the same results (for docs and checks)

```suggestion
            version: Literal["1.0.0", "1.1.1", "1.3.0"] = Query("1.1.1", description="WMS Version"),
```

Reply:
Much cleaner, will remove the enums and change over to literals. 👍